<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  
  <title>gmap使用方法 | ABYSW</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="本文件由abysw软件的blog工具生成和修改，后续将加入添加图片等功能">
<meta property="og:type" content="article">
<meta property="og:title" content="gmap使用方法">
<meta property="og:url" content="https://abysw.github.io/2022/06/06/gmap%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95/index.html">
<meta property="og:site_name" content="ABYSW">
<meta property="og:description" content="本文件由abysw软件的blog工具生成和修改，后续将加入添加图片等功能">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://abysw.github.io/images/swxxfxxx.png">
<meta property="article:published_time" content="2022-06-06T09:16:45.000Z">
<meta property="article:modified_time" content="2022-06-07T02:59:17.700Z">
<meta property="article:author" content="Yuan-SW-F">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://abysw.github.io/images/swxxfxxx.png">
  
    <link rel="alternate" href="/atom.xml" title="ABYSW" type="application/atom+xml">
  
  
    <link rel="shortcut icon" href="/favicon.png">
  
  
    
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/typeface-source-code-pro@0.0.71/index.min.css">

  
  
<link rel="stylesheet" href="/css/style.css">

  
    
<link rel="stylesheet" href="/fancybox/jquery.fancybox.min.css">

  
<meta name="generator" content="Hexo 6.2.0"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">ABYSW</a>
      </h1>
      
        <h2 id="subtitle-wrap">
          <a href="/" id="subtitle">ABYSsWrapper (abysw)</a>
        </h2>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="https://abysw.github.io"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main"><article id="post-gmap使用方法" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2022/06/06/gmap%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95/" class="article-date">
  <time class="dt-published" datetime="2022-06-06T09:16:45.000Z" itemprop="datePublished">2022-06-06</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="p-name article-title" itemprop="headline name">
      gmap使用方法
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p><img src="https://abysw.github.io/images/swxxfxxx.png" alt="使用方法等详细信息请关注公众号【生物信息分析学习】（swxxfxxx）"></p>
<p>【提示】本文写作命令为：abysw blog gmap使用方法</p>
<span id="more"></span>

<p>有很多软件需要反复使用，但是总是会忘了怎么用，需要备注一下。</p>
<p>Consensus sequences, FLNC sequences, and nonredundant transcripts were mapped to the Z. japonica reference transcript sequences (http:&#x2F;&#x2F; zoysia. kazusa. or. jp) using GMAP [50].<br>Then, PacBio sequencing was mapped to the bamboo genome using GMAP with the following option: -B 5 -K 8000 -t 40 -f 2 -n 1 (Wu et al., 2016).</p>
<h5 id="安装gmap"><a href="#安装gmap" class="headerlink" title="安装gmap"></a>安装gmap</h5><p>在maker安装的时候已经安装好了，可以直接用</p>
<ol>
<li><p>建库<br>gmap_build -D .&#x2F; -d ${S}_genome ${S}_genome.fa </p>
</li>
<li><p>比对<br>gmap -B 5 -K 8000 -t 40 -f 2 -n 1  -D .&#x2F; -d ${S}_genome $R.transcripts.hq.fasta &gt; $R.transcripts.hq.fasta.gmap 2&gt; $R.transcripts.hq.fasta.fail</p>
</li>
</ol>
<p>这里我测试的比对率是98.85%。</p>
<p>看看参数：<br>    -B, –batch&#x3D;INT                Batch mode (default &#x3D; 2)<br>    -t, –nthreads&#x3D;INT             Number of worker threads<br>    -f, –format&#x3D;INT               Other format for output (also note the -A and -S options<br>    -n, –npaths&#x3D;INT               Maximum number of paths to show (default 5).  If set to 1, GMAP<br>                                   will not report chimeric alignments, since those imply<br>                                   two paths.  If you want a single alignment plus chimeric<br>                                   alignments, then set this to be 0.<br>                                   上面的三个参数没什么用，-n参数可以调试一下，看看效果。<br>    –min-trimmed-coverage&#x3D;FLOAT   Do not print alignments with trimmed coverage less this value (default&#x3D;0.0, which means no filtering)<br>    –min-identity&#x3D;FLOAT           Do not print alignments with identity less this value (default&#x3D;0.0, which means no filtering)<br>                                   也可以调试这两个参数的用法。</p>
<pre><code>-K 
--max-intronlength-middle=INT  Max length for one internal intron (default 500000).  Note: for backward
                               compatibility, the -K or --intronlength flag will set both
                               --max-intronlength-middle and --max-intronlength-ends.
                               Also see --split-large-introns below.
--max-intronlength-ends=INT    Max length for first or last intron (default 10000).  Note: for backward
                               compatibility, the -K or --intronlength flag will set both
                               --max-intronlength-middle and --max-intronlength-ends.
</code></pre>
<p>&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D; 以下为调试区 &#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;<br>gmap -B 5 -K 8000 -t 40 -f 2 -n 0 -D .&#x2F; -d ${S}_genome $R.transcripts.hq.fasta &gt; $R.transcripts.hq.fasta.gmap0ic 2&gt; $R.transcripts.hq.fasta.gmap.fail0ic &amp;<br>n&#x3D;0 多输出202个嵌合体。</p>
<p>gmap -B 5 -K 8000 -t 40 -f 2 -n 0 –min-trimmed-coverage&#x3D;0.85 –min-identity&#x3D;0.9 -D .&#x2F; -d ${S}_genome $R.transcripts.hq.fasta &gt; $R.transcripts.hq.fasta.gmap0ic 2&gt; $R.transcripts.hq.fasta.gmap.fail0ic &amp;<br>覆盖度和相似度过滤的结果：<br>    未匹配序列：856 –&gt; 1853， 比对率为97.52。<br>    嵌合体序列：170条</p>
<p>gmap –min-trimmed-coverage&#x3D;0.85 –min-identity&#x3D;0.9 -D .&#x2F; -d ${S}_genome $R.transcripts.hq.fasta &gt; $R.transcripts.hq.fasta.gmapic 2&gt; $R.transcripts.hq.fasta.gmap.failic &amp;</p>
<p>测试最基本的参数：<br>gmap -B 5 -t 40 -f 2 -n 1 -D .&#x2F; -d ${S}_genome $R.transcripts.hq.fasta &gt; $R.transcripts.hq.fasta1.gff 2&gt; $R.transcripts.hq.fasta1.gmap.err &amp;<br>gmap -B 5 -t 40 -f 2 -n 0 -D .&#x2F; -d ${S}_genome $R.transcripts.hq.fasta &gt; $R.transcripts.hq.fasta.gff 2&gt; $R.transcripts.hq.fasta.gmap.err &amp;<br>两者的比对率都是98.85%；<br>嵌合体为213个（较之前多11个嵌合体，为-K参数所致）。<br>path2 暂时不知道有没有用，先用文献中的参数吧。<br>部分基因跨度太大。<br>&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;<br>gmap_build -D .&#x2F; -d ${S}_genome ${S}_genome.fa<br>gmap -B 5 -K 8000 -t 40 -f 2 -n 1 -D .&#x2F; -d ${S}_genome $R.transcripts.hq.fasta &gt; $R.transcripts.hq.fasta.gff 2&gt; $R.transcripts.hq.fasta.gmap.err &amp;</p>
<p>#################文件默认变量#####################<br>$G genome fasta file<br>$S species<br>$R NGS reads &#x2F; NGS sample name<br>##################################################</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://abysw.github.io/2022/06/06/gmap%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95/" data-id="cl5qfvpiu000am6bzg6hi61pp" data-title="gmap使用方法" class="article-share-link">Share</a>
      
      
      
    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2022/06/08/APA%E5%88%86%E6%9E%90TAPIS/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          APA分析TAPIS
        
      </div>
    </a>
  
  
    <a href="/2022/06/04/mysql%E9%87%8D%E5%90%AF%E6%9C%8D%E5%8A%A1%E5%99%A8%E5%90%8E%E5%86%8D%E6%AC%A1%E5%A4%B1%E6%95%88/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">mysql重启服务器后再次失效</div>
    </a>
  
</nav>

  
</article>


</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Categories</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/%E5%8D%9A%E5%AE%A2/">博客</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/%E5%8D%9A%E5%AE%A2%E5%86%99%E4%BD%9C/">博客写作</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/%E6%9C%8D%E5%8A%A1%E5%99%A8/">服务器</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/%E6%B5%8B%E8%AF%95/">测试</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/%E6%B5%8B%E8%AF%95/%E5%8D%9A%E5%AE%A2/">博客</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/%E7%94%9F%E4%BF%A1/">生信</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/%E7%94%9F%E4%BF%A1%E5%88%86%E6%9E%90/">生信分析</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tags</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/%E5%86%99%E4%BD%9C%E6%B5%8B%E8%AF%95/" rel="tag">写作测试</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/%E5%8D%9A%E5%AE%A2%E5%86%99%E4%BD%9C/" rel="tag">博客写作</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/%E5%8D%9A%E5%AE%A2%E6%B5%8B%E8%AF%95/" rel="tag">博客测试</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A/" rel="tag">基因注释</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/%E6%9C%8D%E5%8A%A1%E5%99%A8%E7%BB%B4%E6%8A%A4/" rel="tag">服务器维护</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/%E6%B3%A8%E9%87%8A/" rel="tag">注释</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tag Cloud</h3>
    <div class="widget tagcloud">
      <a href="/tags/%E5%86%99%E4%BD%9C%E6%B5%8B%E8%AF%95/" style="font-size: 10px;">写作测试</a> <a href="/tags/%E5%8D%9A%E5%AE%A2%E5%86%99%E4%BD%9C/" style="font-size: 20px;">博客写作</a> <a href="/tags/%E5%8D%9A%E5%AE%A2%E6%B5%8B%E8%AF%95/" style="font-size: 10px;">博客测试</a> <a href="/tags/%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A/" style="font-size: 10px;">基因注释</a> <a href="/tags/%E6%9C%8D%E5%8A%A1%E5%99%A8%E7%BB%B4%E6%8A%A4/" style="font-size: 10px;">服务器维护</a> <a href="/tags/%E6%B3%A8%E9%87%8A/" style="font-size: 10px;">注释</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/07/">July 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/06/">June 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/05/">May 2022</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/07/18/update/">update</a>
          </li>
        
          <li>
            <a href="/2022/07/18/%E8%BF%9C%E7%A8%8B%E6%A1%8C%E9%9D%A2%E8%AE%BF%E9%97%AElinux/">远程桌面访问linux</a>
          </li>
        
          <li>
            <a href="/2022/07/08/%E5%9F%BA%E5%9B%A0%E5%8A%9F%E8%83%BD%E6%B3%A8%E9%87%8A%E7%AC%94%E8%AE%B0/">基因功能注释笔记</a>
          </li>
        
          <li>
            <a href="/2022/06/20/arm%E6%9C%8D%E5%8A%A1%E5%99%A8%E5%AE%89%E8%A3%85samtools/">arm服务器安装samtools</a>
          </li>
        
          <li>
            <a href="/2022/06/17/%E9%80%9A%E8%BF%87%E5%B1%80%E5%9F%9F%E7%BD%91%E5%85%B1%E4%BA%AB%E7%BD%91%E7%BB%9C%E8%BF%9E%E6%8E%A5%EF%BC%88%E6%B5%8B%E8%AF%95%E7%89%88%EF%BC%89/">通过局域网共享网络连接（测试版）</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      
      &copy; 2022 Yuan-SW-F<br>
      Powered by <a href="https://abysw.github.io/" target="_blank">ABYSW</a>
    </div>
  </div>
</footer>

    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    


<script src="/js/jquery-3.4.1.min.js"></script>



  
<script src="/fancybox/jquery.fancybox.min.js"></script>




<script src="/js/script.js"></script>





  </div>
</body>
</html>